dix/req.c

```diff
@@ -1620,29 +1641,69 @@ static void dix_cas_rop_rc_update(struct m0_dix_cas_rop *cas_rop, int rc)
 
 static void dix_rop_completed(struct m0_sm_group *grp, struct m0_sm_ast *ast)
 {
-	struct m0_dix_req     *req = ast->sa_datum;
-	struct m0_dix_rop_ctx *rop = req->dr_rop;
-	struct m0_dix_rop_ctx *rop_del_phase2 = NULL;
-	bool                   del_phase2 = false;
-	struct m0_dix_cas_rop *cas_rop;
+	struct m0_dix_req      *req = ast->sa_datum;
+	struct m0_dix_rop_ctx  *rop = req->dr_rop;
+	struct m0_dix_rop_ctx  *rop_del_phase2 = NULL;
+	bool                    del_phase2 = false;
+	struct m0_dix_cas_rop  *cas_rop;
+	struct m0_pool_version *pver;
+	int64_t                 min_success;
+	int64_t                 successful_ops = 0;
 
 	(void)grp;
 	if (req->dr_type == DIX_NEXT)
 		m0_dix_next_result_prepare(req);
 	else {
+		min_success = req->dr_min_success;
+		M0_ASSERT(ergo(min_success < 1,
+			       min_success == M0_DIX_MIN_REPLICA_QUORUM));
+		if (min_success == M0_DIX_MIN_REPLICA_QUORUM) {
+			pver = m0_dix_pver(req->dr_cli, &req->dr_indices[0]);
+			min_success = (pver->pv_attr.pa_N +
+				       pver->pv_attr.pa_K)/2 + 1;
+		}
+		M0_ASSERT(min_success > 0);
```

Comment:
It would make more sense to handle M0_DIX_MIN_REPLICA_QUORUM when the operation is initialised, rather than when it is completed. But if doing it during initialisation is somehow difficult, it's fine to do it here.

Reply:
Agreed, but finding N and K with `m0_dix_pver` looks like it requires layout info (which doesn't seem to be available during init, need to wait until op launch, correct?)

Reply:
I think in dix_rop_ctx_init() we can get the values of N and K from DIX pool version.